src/dcqc/parsers.py

```diff
@@ -45,10 +52,83 @@ def create_suites(
         skipped_tests: Optional[Collection[str]] = None,
     ) -> Iterator[SuiteABC]:
         for target in self.create_targets():
-            # This function assumes that there is one file per target,
-            # which is always the case when parsing CSV files (for now)
-            first_file = target.files[0]
-            file_type = first_file.get_file_type()
-            suite_cls = SuiteABC.get_subclass_by_file_type(file_type)
-            suite = suite_cls(target, required_tests, skipped_tests)
-            yield suite
+            yield SuiteABC.from_target(target, required_tests, skipped_tests)
+
+
+class JsonParser:
+    path: Path
+
+    def __init__(self, path: Path):
+        self.path = path
+
+    @classmethod
+    def parse_expected(cls, path: Path, expected_cls: Type[T]) -> T:
+        """Generate expected object from JSON file.
+
+        Args:
+            json_path: JSON file describine the expected object.
+
+        Raises:
+            ValueError: If the JSON file does not
+                describe a expected object.
+
+        Returns:
+            The reconstructed object.
+        """
+        parser = JsonParser(path)
+        object_ = parser.parse_object()
```

Comment:
Good point. `parse_expected()` came afterwards. I refactored the class so that `parse_object()` and `parse_objects()` both have an `expected_cls` parameter and removed the `parse_expected()` method.